cloudslang-compiler/src/main/java/io/cloudslang/lang/compiler/SlangCompilerImpl.java

```diff
@@ -127,7 +127,7 @@ public ExecutableModellingResult preCompileSource(SlangSource source) {
     }
 
     private ParsedSlang parseSystemPropertiesFile(SlangSource source) {
-        ParsedSlang parsedSlang = yamlParser.parse(source);
+        ParsedSlang parsedSlang = yamlParser.parseAndValidateAndThrowFirstError(source);
```

Comment:
seems that different it would be more generic to be able to wire in the interface method a strategy to validate, for example proceed_on_failure (this strategy collects all validation errors and message), fail_on_first (thorws exception on first invalid thing)